relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -1,20 +1,137 @@
 use std::collections::BTreeMap;
+use std::fmt::{self, Display};
+use std::str::FromStr;
 
-use relay_common::{SpanStatus, UnixTimestamp};
+use relay_common::{MetricUnit, SpanStatus, UnixTimestamp};
 use relay_dynamic_config::{AcceptTransactionNames, TaggingRule, TransactionMetricsConfig};
 use relay_general::protocol::{
     AsPair, Context, ContextInner, Event, EventType, TraceContext, TransactionSource, User,
 };
 use relay_general::store;
 use relay_general::types::Annotated;
 use relay_metrics::{
-    AggregatorConfig, DurationUnit, Metric, MetricNamespace, MetricUnit, MetricValue,
+    AggregatorConfig, DurationUnit, Metric, MetricNamespace, MetricResourceIdentifier, MetricType,
+    MetricValue, ParseMetricError,
 };
 
 use crate::metrics_extraction::conditional_tagging::run_conditional_tagging;
 use crate::statsd::RelayCounters;
 use crate::utils::SamplingResult;
 
+/// Enumerates the most common transaction-names, with a catch-all for any other names.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum TransactionsKind {
+    User,
+    Duration(DurationUnit),
+    CountPerRootProject,
+    Breakdowns(String),
+    Measurements {
+        kind: Measurementkind,
+        unit: MetricUnit,
+    },
```

Comment:
I was wrong about measurements before. They are dynamically configured and validated by event normalization: https://github.com/getsentry/relay/blob/73ea5adc59ec80f9eb300d7717441f6262b54bc2/relay-general/src/store/normalize.rs#L352.

What this means is we do not need / should not have a `MeasurementKind`, and just use `Measurements(String, MetricUnit)` to get the name and the unit.